MyXamarinApp.API/Controllers/EmployeeController.cs

```diff
@@ -17,25 +20,112 @@ public EmployeeController(IEmployeeRepository employeeRepository)
             _employeeRepository = employeeRepository;
         }
 
+
         [HttpGet]
-        public async Task<ActionResult<List<EmployeeModel>>> GetAllEmployees()
+        public async Task<ActionResult<List<EmployeeModel>>> GetAllEmployees([FromQuery] string searchString, DateTime initialDate, DateTime finalDate, int currentPage = 1, int pageSize = 10, bool status = true)
         {
-            var employees = await _employeeRepository.GetAllEmployees();
-            return employees;
+            try
+            {
+                var employees = await _employeeRepository.GetAllEmployees(searchString, initialDate, finalDate, status, currentPage, pageSize);
+                return Ok(employees);
+            }
+            catch (Exception ex)
+            {
+                return new StatusCodeResult(StatusCodes.Status500InternalServerError);
+            }
+        }
+
+        [HttpGet("{id}")]
+        public async Task<ActionResult<List<EmployeeModel>>> GetEmployeeById(Guid id)
```

Comment:
If you want to you can also include a validation check for an empty guid, that will just save the the hit to the database ?